src/group_theory/specific_groups/alternating.lean

```diff
@@ -125,8 +137,88 @@ lemma is_three_cycle_is_conj (h5 : 5 ≤ fintype.card α)
 alternating_group.is_conj_of (is_conj_iff_cycle_type_eq.2 (hσ.trans hτ.symm))
   (by rwa hσ.card_support)
 
+
+lemma three_cycle_is_commutator
+  (h5: 5 ≤ fintype.card α) {g : alternating_group α} :
+   is_three_cycle (g:perm α)
+   → ∃ (p q : alternating_group α), g = p * q * p⁻¹ * q⁻¹  :=
+begin
+  intro hg,
+    -- g^2 est conjugué de g : g^2 = h g h⁻¹ , donc hg = [h,g]
+  let hg2 := hg.is_three_cycle_sq, -- g^2 is a 3-cycle
+  rw ← subgroup.coe_mul _ g g at hg2,
+  obtain ⟨h, hh⟩ := alternating_group.is_three_cycle_is_conj h5 hg hg2,
+  -- g^2 is conjugate to g ;  hh :  g^2 * h = h * g
+  have : g = h * g * h⁻¹ * g⁻¹ ,
+  exact calc g = (g * g * ↑h) * (↑h)⁻¹ * g ⁻¹ : by simp
+    ... = (↑h * g) * (↑h)⁻¹  * g⁻¹  : by rw hh.symm
+    ... = ↑h * g * (↑h)⁻¹ * g⁻¹ : by simp,
+  exact ⟨h, g, this⟩,
+  end
+
+lemma three_cycle_mem_commutator
+  (h5: 5 ≤ fintype.card α) {g : alternating_group α} :
+   is_three_cycle (g:perm α)
+   → g ∈ commutator (alternating_group α)  :=
+begin
+  intro hg,
+  apply subgroup.subset_normal_closure ,
+  rw set.mem_set_of_eq,
+  obtain ⟨p, q, hpq⟩ := three_cycle_is_commutator h5 hg,
+  use ⟨p, q, hpq.symm⟩
+end
+
 end alternating_group
 
+section ugly_lemmas
+private lemma ugly_lemma (A : Type*) [group A] (B : subgroup A) (p : A → Prop) (hp : ∀(a : A), p a → a ∈ B) :
+  subgroup.comap B.subtype (subgroup.closure ({ a : A | p a }))
+    =  (subgroup.closure({ b : B | p ↑b})) :=
+begin
+  ext ⟨a, ha⟩,
+  simp,
+  split,
+  { intro hpa,
+    apply mem_closure.mpr,
+    intros K hpb,
+    have : set_of p ⊆ (subgroup.map B.subtype K),
+    { intros x hx, rw set.mem_set_of_eq at hx,
+      simp only [subgroup.coe_subtype, set.mem_image, subgroup.coe_map, set_like.mem_coe],
+      rw set_like.exists,
+      use x,
+      use hp x hx,
+      split,
+      apply hpb,
+      rw set.mem_set_of_eq, exact hx,
+      exact subtype.coe_mk x (hp x hx) },
+    obtain ⟨x, H, hH⟩ := mem_closure.mp hpa _ this,
+    rw [subgroup.coe_subtype, ← set_like.coe_mk a ha,  subgroup.coe_mk,
+      ← set_like.coe_mk a ha, set_like.coe_eq_coe ] at hH,
+    rw ← hH, exact H },
+  { intro h,
+    apply mem_closure.mpr,
+    intros K hpK,
+    have hpKB : { b : ↥B | p ↑b } ⊆ subgroup.comap B.subtype K,
+    { intros b hb,
+      rw set.mem_set_of_eq at hb,
+      exact hpK hb },
+    exact subgroup.mem_comap.mp (mem_closure.mp h (subgroup.comap B.subtype K) hpKB),
+  }
+end
+
+private lemma stupid_lemma {A : Type*} [group A] (B : subgroup A) :
```

Comment:
This should be called `subgroup.comap_subtype_self` to match `submodule.comap_subtype_self`,  and should go in `subgroup/basic.lean`.